var/spack/repos/builtin/packages/libffi/package.py

```diff
@@ -32,6 +32,7 @@ class Libffi(AutotoolsPackage):
     patch("clang-powerpc-3.2.1.patch", when="@3.2.1%clang platform=linux")
     # ref.: https://github.com/libffi/libffi/pull/561
     patch("powerpc-3.3.patch", when="@3.3")
+    patch("forward_declare_open_temp_exec_file.patch", when="@3.4.3:3.4.4%clang@15:")
```

Comment:
Could you change this to use the patch _directly_ from github instead (see other uses of `patch("https://github.com/...patch?full_index=1, ...) for examples)? That way it's clear that it's available upstream and one can find the context for the patch as well through the same link.

Reply:
The patch comes from here: https://github.com/libffi/libffi/pull/764.

Reply:
Somewhat related, is it maybe worth it to apply the patch independently of the compiler? It should never hurt to apply that patch, and if e.g. GCC or NVHPC makes it a hard error as well in a future version it'll already be covered.